source/common/vb_gui.c

```diff
@@ -343,14 +347,38 @@ int emulation_reset(void) {
     return D_EXIT;
 }
 
-int emulation_sstate(void) {
+char * get_savestate_path(int state) {
+    char * sspath;
+    sspath = (char *) malloc(131 * sizeof(char));
+    char file_name[8];
+    char * last_slash = strrchr(tVBOpt.ROM_PATH, '/');
+    strncpy(sspath, tVBOpt.ROM_PATH, last_slash - tVBOpt.ROM_PATH);
+    sspath[last_slash - tVBOpt.ROM_PATH] = '\0';
+    strcat(sspath, "/vb_savestates");
+    if(stat(sspath, &st) == -1)
+        mkdir(sspath, 0777);
+    strcat(sspath, tVBOpt.ROM_PATH + (last_slash - tVBOpt.ROM_PATH));
+    sspath[strlen(sspath) - 3] = '\0';
+    if(stat(sspath, &st) == -1)
+        mkdir(sspath, 0777);
+    sprintf(file_name, "/%d.rds", state);
+    strcat(sspath, file_name);
+
+    return sspath;
+}
+
+int emulation_rmstate(int state) {
+    char* sspath = get_savestate_path(state);
```

Comment:
When will the memory be released?

Reply:
addressed in [9f26379](https://github.com/skyfloogle/red-viper/pull/59/commits/9f26379e5c1fc95c22ec9355840a60331aff69de)